back-end/helpers/intent_analyzer.js

```diff
@@ -0,0 +1,56 @@
+export default class IntentAnalyzer{
+    /* What it should result in:
+    [
+        {
+          question: "What is something that Waluigi and Wario often say?",
+          children: {
+            "The brothers Waluigi and Wario have their catchphrase 'waaa'.": 100,
+          }
+        },
+        {
+          question: "Where should I go for a good time and great learning experience?",
+          children: {
+            "You should go to the University of Toronto.": 10,
+            "You should go to the Technology Leadership Initiative.": 90
+          }
+        }, 
+        {
+          question: "Hey BestFlow, what time is it?",
+          children: {
+            "It's time for you to get a watch, would you like to take a look at our product?": 10,
+            "Showtime showtime what I'm John Laurens in the place to be.": 20,
+            "Summertime! School's out, scream and shout.": 70
+          }
+        }
+    ]
+      */
+
+    static async formatIntents(intents) { // Formats the mongoose Model into desired array for IntentLister to use
+      if (Object.keys(intents).length == 0) {
+        return {};
+      }
+
+      var percentageMap;
+      if (intents.total_children == 0) {
+        percentageMap = {"No intents found.": 0};
```

Comment:
I'm wondering if there is a better way of handling there being no children. Ideally on the front-end we don't want to just display "No intents found." with 0% as if it were an actual intent, since this is more of a special case. The only way I can imagine handling this right now is checking for an exact match of the intent and percentage, which gives a (very small) possibility that this conflicts with an actual intent with the same string and percent. Maybe overly pedantic so someone else confirm?

Reply:
Yea, when I was writing this I was trying to think of a way that would make it clear that it isn't an actual intent, however I'm just wondering how IntentLister would want to handle it, so for now I'm thinking to leave it like this until we find a more concrete solution, I wonder if @helias22 would know if there's anything about the data that makes it so there's never empty children or if there is, if she formatted them in a special way?

Reply:
Right now it's just saving the number of children. So if there is no children the number is 0. And that's a situation that might happen. But I'm not sure how we want to display that to avoid any misunderstanding.